### PR TITLE
Option to export zone with objects embedded

### PR DESCRIPTION
**Pull request by knervous**
Sorry for the big PR here, lots of formatting changes too and I ended up with some bugs along the way that I ended up fixing and didn't want to create multiple PRs for, but if that makes things easier I can do that too.

Generally what this should be doing is checking for the new flag in settings ExportZoneWithObjects and if it's present, down in ExportZone we try to grab associated object info and pack in the meshes and each instance of the object within the zone and apply coordinate offset, scale and rotation to each fragment.

Some other things I included here were:
- Null checking on ghost ship, something in my EQ dir was blowing up there
- Allow chunking of batching up exports which helps memory not get flooded. Might only be an issue for obj export.
- z coordinate was sometimes coming up FF FF and overflowing to -32769 based on floating point precision? Not sure exactly, but put a check in for that
- Lots of formatting, I blame my Visual Studio sorry about that!

### CODE REVIEW

File: LanternExtractor/EQ/Wld/Fragments/ObjectInstance.cs [hunk 1]
```diff
@@ -55,8 +56,16 @@ class ObjectInstance : WldFragment
             // In main zone, it points to a 0x16 fragment
             // In objects.wld, it is 0
             int unknown2 = Reader.ReadInt32();
+            var x = Reader.ReadSingle();
+            var y = Reader.ReadSingle();
+            var z = Reader.ReadSingle();
 
-            Position = new vec3(Reader.ReadSingle(), Reader.ReadSingle(), Reader.ReadSingle());
+            // Sometimes we are getting the lowest value of signed int 16. Reset to 0 otherwise this z will be way below the world. Round for floating point errors.
+            if (Math.Round(z) <= short.MinValue)
```
Comment (danwilkins) [inline LanternExtractor/EQ/Wld/Fragments/ObjectInstance.cs]: This change shouldn't be in this fragment. My guess is that they used a "dropping" solution where they place a bunch of trees and dropped them onto the terrain below. Some of them missed and ended up at the bottom of the world. So the issue here is that these trees will now be positioned at z = 0. What I'd recommend is filtering them out in the export code. I do it in the importer in a rather lazy way. `if(z < 30000) continue;`

Can definitely handle it more elegantly.

File: LanternExtractor/EQ/Wld/Helpers/CharacterFixer.cs [hunk 2]
```diff
@@ -281,11 +281,11 @@ private void FixShipNames()
 
             foreach (var actor in actors)
             {
-                if (actor.Name.StartsWith("GSP"))
+                if (actor.Name.StartsWith("GSP") && (actor?.MeshReference?.Mesh?.Name ?? null) != null)
```
Comment (danwilkins) [inline LanternExtractor/EQ/Wld/Helpers/CharacterFixer.cs:284]: Good catch. Must be a Titanium issue.

**Review by danwilkins — CHANGES_REQUESTED** [review]

File: LanternExtractor/EQ/Wld/Fragments/ObjectInstance.cs (hunk 1 above)
Comment (knervous) [reply LanternExtractor/EQ/Wld/Fragments/ObjectInstance.cs]: Good call here nobody else should care about dropped trees here, moved this to MeshObjWriter

File: LanternExtractor/packages.config [hunk 3]
```diff
@@ -4,6 +4,7 @@
   <package id="DotNetZip" version="1.10.1" targetFramework="net452" />
   <package id="Fody" version="6.4.0" targetFramework="net48" developmentDependency="true" />
   <package id="GlmSharp" version="0.9.8" targetFramework="net452" />
+  <package id="Ionic.Zip" version="1.9.1.8" targetFramework="net48" />
```
Comment (nickgal) [inline LanternExtractor/packages.config]: Is this needed or used in addition to `DotNetZip`?

Comment (knervous) [reply LanternExtractor/packages.config]: No idea how that guy got installed, good catch. Probably visual studio being smart about something. Removed

**Review by danwilkins — APPROVED** [review]

**Pull request merged**